src/tiled/command.cpp

```diff
@@ -34,56 +34,58 @@ QString Command::finalWorkingDirectory() const
 {
     QString finalWorkingDirectory = workingDirectory;
 
-    // Perform variable replacement
-    if (Document *document = DocumentManager::instance()->currentDocument()) {
-        const QString fileName = document->fileName();
-
-        QFileInfo fileInfo(fileName);
-        QString mapPath = fileInfo.absolutePath();
-        finalWorkingDirectory.replace(
-            QLatin1String("%mappath"),
-            QString(QLatin1String("%1")).arg(mapPath));
-        // QProcess::setWorkingDirectory doesn't work with quoted paths
-
-        // TODO: %executablepath
-    }
-
-    return finalWorkingDirectory;
+    return replaceVariables(finalWorkingDirectory, true);
 }
 
 QString Command::finalCommand() const
 {
     QString finalCommand = QString(QLatin1String("%1 %2")).arg(executable).arg(arguments);
 
+    return replaceVariables(finalCommand);
+}
+
+QString Command::replaceVariables(const QString &string, bool flag) const
+{
+    QString finalString = string;
+
     // Perform variable replacement
     if (Document *document = DocumentManager::instance()->currentDocument()) {
         const QString fileName = document->fileName();
 
-        finalCommand.replace(QLatin1String("%mapfile"),
+        finalString.replace(QLatin1String("%mapfile"),
                              QString(QLatin1String("\"%1\"")).arg(fileName));
 
         QFileInfo fileInfo(fileName);
         QString mapPath = fileInfo.absolutePath();
-        finalCommand.replace(
-            QLatin1String("%mappath"),
-            QString(QLatin1String("\"%1\"")).arg(mapPath));
+
+        if (flag) {
+            finalString.replace(
+                QLatin1String("%mappath"),
+                QString(QLatin1String("%1")).arg(mapPath));
+        } else {
+            finalString.replace(
+                QLatin1String("%mappath"),
+                QString(QLatin1String("\"%1\"")).arg(mapPath));
+        }
```

Comment:
I think this does not really solve the problem. My idea with having a single function for replacement of the variables was that each variable can be used in both (the command and the working dir) settings. Here you only do special handling for this specific variable. But extending this approach to the other variables does not really fly either i think... If we wrap each replacement with an `if` we could simply stick to two functions as before. So I think this is better solved by doing a final replacement of `"` characters with an empty string (after all the previous variables are replaced) maybe?

@bjorn: any opinions here?